addon/mixins/model-records/relationships.js

```diff
@@ -19,25 +19,25 @@ function relationshipMacro(type) {
       .filter(group => group.name === type)
       .reduce((relationships, group) => {
         group.properties.forEach(property => {
-          let records = get(this, `model.${property}`)
+          get(this, `model.${property}`).then(records => {
```

Comment:
I think it is partially what is causing the weird loading.  Since we are now using nested routes, I am not quite sure why we need this entirely.  This prefetches all the relationship records and all we need is a list of the types so that we can construct the `route`.  We then just load the model as usual in the router